### PR TITLE
Document the web dashboard feature

### PR DESCRIPTION
**Pull request by oleiade**
## What?

This PR aims to document the web dashboard feature that lands in v0.49.0 of k6.

I've been struggling somewhat to put this together, and I'd really like as much input, suggestions and feedback as possible as to how to structure this, the amount of detail we want to provide, as well as where it should live. 

@szkiba I would especially love your input on this too 👍

## Checklist

Please fill in this template:
- [X] I have used a meaningful title for the PR.
- [X] I have described the changes I've made in the "What?" section above.
- [X] I have performed a self-review of my changes.
- [X] I have run the `make docs` command locally and verified that the changes look good.
- [X] I have made my changes in the `docs/sources/next` folder of the documentation.

### CODE REVIEW

File: docs/sources/next/results-output/web-dashboard/_index.md [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+The web dashboard can be configured using environment variables:
+
+| Environment variable               | Description                                   | Default value |
+| ---------------------------------- | --------------------------------------------- | ------------- |
+| `K6_WEB_DASHBOARD`                 | Enable the web dashboard                      | `false`       |
+| `K6_WEB_DASHBOARD_HOST`            | Host to bind the web dashboard to             | `localhost`   |
+| `K6_WEB_DASHBOARD_PORT`            | Port to bind the web dashboard to             | `5665`        |
+| `K6_WEB_DASHBOARD_PERIOD`          | Period in seconds to update the web dashboard | `10s`         |
+| `K6_WEB_DASHBOARD_OPEN`            | Open the web dashboard in the default browser | `false`       |
+| `K6_WEB_DASHBOARD_REPORT_FILENAME` | Filename to export the HTML test report to    | `report.html` |
```
Comment (szkiba) [inline docs/sources/next/results-output/web-dashboard/_index.md:59]: `K6_WEB_DASHBOARD_EXPORT` is the exported report file name env var (not `K6_WEB_DASHBOARD_REPORT_FILENAME`)

File: docs/sources/next/results-output/web-dashboard/_index.md [hunk 2]
```diff
@@ -0,0 +1,59 @@
...
+## HTML test reports
+
+The dashboard enables to generate detailed, downloadable HTML reports directly from the dashboard. These reports are self-contained, making them ideal for sharing with your team.
+
+![HTML test report screenshot](/media/docs/k6-oss/web-dashboard-report.png)
+
+The HTML test report can be accessed from the dashboard's menu, by clicking on the "Report" button.
+
+![HTML test report generation button](/media/docs/k6-oss/web-dashboard-report-button.png)
+
```
Comment (szkiba) [inline docs/sources/next/results-output/web-dashboard/_index.md:26]: Perhaps a sentence could be written about how the report can be automatically saved using the `K6_WEB_DASHBOARD_EXPORT` environment variable, without starting a browser.

**Review by szkiba — CHANGES_REQUESTED** [review]
`K6_WEB_DASHBOARD_REPORT_FILENAME` should be change to `K6_WEB_DASHBOARD_EXPORT`

File: docs/sources/next/results-output/web-dashboard/_index.md [hunk 3]
```diff
@@ -0,0 +1,59 @@
...
+## How to use
```
Comment (heitortsergent) [inline docs/sources/next/results-output/web-dashboard/_index.md:27]: I think we should move this section above the "HTML test reports" one, so it goes: Introduction, How to use, HTML reports, Dashboard options.

File: docs/sources/next/results-output/web-dashboard/_index.md [hunk 4]
```diff
@@ -0,0 +1,59 @@
...
+
+The dashboard provides a real-time overview of the performance observed by k6 while a test is running, and helps to identify potential reliability issues as they occur.
+
+## HTML test reports
+
+The dashboard enables to generate detailed, downloadable HTML reports directly from the dashboard. These reports are self-contained, making them ideal for sharing with your team.
+
+![HTML test report screenshot](/media/docs/k6-oss/web-dashboard-report.png)
+
+The HTML test report can be accessed from the dashboard's menu, by clicking on the "Report" button.
```
Comment (heitortsergent) [inline docs/sources/next/results-output/web-dashboard/_index.md:23]:
```suggestion
You can access the HTML test report by clicking the "Report" button on the dashboard's menu.
```

**Review by heitortsergent — COMMENTED** [review]
@oleiade I think this is really good, thanks for writing this up!

I made some suggestions for changing the order of the sections and the headings, and some suggestions related to our Grafana style guide. :)

I think the location of the page makes sense, we could also add a mention of it to the "Results output" page under the Get started section. Maybe as a new heading right above the "Time series and external outputs" section: https://grafana.com/docs/k6/latest/get-started/results-output/#time-series-and-external-outputs

Two questions I have:

- Would it make sense to have some sort of admonition at the top and point people to the xk6-dashboard repository if people want to give feedback on the dashboard?
- I saw this section on the xk6-dashboard README, and I'm wondering if we should include it as well?

   "The k6 process waits to exit as long as there is at least one open browser window for the dashboard extension. In this way, the report can be downloaded, for example, even after the test has been completed.

   In certain environments, it is not allowed that the k6 process does not exit after the test run (eg CI/CD pipeline). In this case, it is advisable to disable the HTTP port (with the -1 value of port parameter)."

File: docs/sources/next/results-output/web-dashboard/_index.md [hunk 5]
```diff
@@ -0,0 +1,59 @@
...
+
+# Web dashboard
+
+k6 provides a built-in web dashboard running on the user's machine and enabling them to monitor their test results in real-time.
+
+## Real-time test results
+
+![Web dashboard screenshot](/media/docs/k6-oss/web-dashboard-overview.png)
+
+The dashboard provides a real-time overview of the performance observed by k6 while a test is running, and helps to identify potential reliability issues as they occur.
```
[the remark this answers is not in the document]
Comment (oleiade) [reply docs/sources/next/results-output/web-dashboard/_index.md:15]: Sounds good 👍🏻

**Review by szkiba — APPROVED** [review]

File: docs/sources/next/results-output/web-dashboard/_index.md [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+
+## Generate HTML test reports
+
+You can generate detailed, downloadable HTML reports directly from the dashboard. These reports are self-contained, making them ideal for sharing with your team.
+
+![HTML test report screenshot](/media/docs/k6-oss/web-dashboard-report.png)
+
+You can access the HTML test report by clicking the "Report" button on the dashboard's menu.
+
+![HTML test report generation button](/media/docs/k6-oss/web-dashboard-report-button.png)
```
Comment (heitortsergent) [inline docs/sources/next/results-output/web-dashboard/_index.md:23]:
````suggestion
You can generate detailed, downloadable HTML reports directly from the web dashboard or the command line. These reports are self-contained, making them ideal for sharing with your team.

![HTML test report screenshot](/media/docs/k6-oss/web-dashboard-report.png)

### Generate report from web dashboard

To generate a report from the web dashboard, click **Report** on the dashboard's menu.

![HTML test report generation button](/media/docs/k6-oss/web-dashboard-report-button.png)

### Generate report from the command line

To automatically generate a report from the command line once the test finishes running, use the `K6_WEB_DASHBOARD_EXPORT` option. For example:

```shell
K6_WEB_DASHBOARD=true K6_WEB_DASHBOARD_EXPORT=html-report.html k6 run script.js
```
````

@oleiade just a suggestion here taking into account @szkiba feedback to mention `K6_WEB_DASHBOARD_EXPORT`.

**Review by heitortsergent — APPROVED** [review]
@oleiade thanks for making the updates!

I think we can get this merged if you want and make improvements later on. :)

I'd still like to move the "How to use" section up, and I left a suggestion for how we can mention the `K6_WEB_DASHBOARD_EXPORT` variable.

@szkiba I don't know if you saw my comment, but are the warnings we have on the [xk6-dashboard Usage section](https://github.com/grafana/xk6-dashboard?tab=readme-ov-file#usage) still applicable here? If they are, I think it would make sense to mention it in the docs as well.

**Pull request merged**